crates/common/trie/trie.rs

```diff
@@ -262,21 +262,17 @@ impl Trie {
             .map(|node| {
                 (
                     NodeHash::from_slice(&Keccak256::new_with_prefix(node).finalize()),
-                    node,
+                    node.clone(),
                 )
             })
             .collect::<HashMap<_, _>>();
```

Comment:
Is there any reason to not pass ownership of the nodes to `from_nodes` so you can avoid the clone altogether?